modules/reactor/src/meshgenerators/DepletionIDGenerator.C

```diff
@@ -0,0 +1,114 @@
+DepletionIDGenerator::generate()
+{
+  std::unique_ptr<MeshBase> mesh = std::move(_input);
+  // parsing the extra ids
+  std::vector<ExtraElementIDName> id_names;
+  id_names = getParam<std::vector<ExtraElementIDName>>("id_name");
+  if (!mesh->has_elem_integer(_material_id_name))
+    mooseError("material_id is not defined in input mesh!");
+  id_names.push_back(_material_id_name);
+  std::set<SubdomainID> block_ids = {Moose::ANY_BLOCK_ID};
+  std::map<dof_id_type, dof_id_type> parsed_ids =
+      ExtraIDUtils::ParseExtraIDs(*mesh, block_ids, id_names);
+  // re-numbering if exclude_id_name is used
+  if (isParamValid("exclude_id_name") && isParamValid("exclude_id_value"))
+  {
+    const auto exclude_id_name = getParam<std::vector<ExtraElementIDName>>("exclude_id_name");
+    const auto exclude_id_value =
+        getParam<std::vector<std::vector<dof_id_type>>>("exclude_id_value");
+    std::vector<unsigned int> id_index;
+    std::vector<std::set<dof_id_type>> id_value_set;
+    id_index.resize(exclude_id_name.size());
+    id_value_set.resize(exclude_id_name.size());
+    for (unsigned int i = 0; i < exclude_id_name.size(); ++i)
+    {
+      id_index[i] = mesh->get_elem_integer_index(exclude_id_name[i]);
+      std::copy(exclude_id_value[i].begin(),
+                exclude_id_value[i].end(),
+                std::inserter(id_value_set[i], id_value_set[i].end()));
+    }
+    // list of unique IDs not removed by "exclude_id_name" and "exclude_id_value" option
+    std::set<dof_id_type> ids;
+    for (const auto & elem : mesh->active_local_element_ptr_range())
+    {
+      // don't need to check if this unique parsed ID corresponding to this element is already in
+      // the "ids"
+      if (ids.count(parsed_ids[elem->id()]))
+        continue;
+      // check extra IDs of this element is defined in  "exclude_id_name" and "exclude_id_value"
+      bool is_exclude_elem = false;
+      for (unsigned int i = 0; i < id_index.size(); ++i)
+      {
+        const auto id = elem->get_extra_integer(id_index[i]);
+        if (id_value_set[i].count(id))
+        {
+          is_exclude_elem = true;
+          break;
+        }
+      }
+      // if this element does not need to be excluded, having unique ID, add to "ids"
+      if (!is_exclude_elem)
+        ids.insert(parsed_ids[elem->id()]);
+    }
+    comm().set_union(ids);
+    // reassign parsed (depletion) ids
+    for (const auto & elem : mesh->active_local_element_ptr_range())
+    {
+      dof_id_type id = parsed_ids[elem->id()];
+      dof_id_type new_id = 0;
+      if (ids.count(id))
+        new_id = std::distance(ids.begin(), std::find(ids.begin(), ids.end(), id)) + 1;
+      parsed_ids[elem->id()] = new_id;
+    }
+  }
+  // assign depletion id to mesh
+  unsigned int depletion_id;
+  if (mesh->has_elem_integer("depletion_id"))
+    depletion_id = mesh->get_elem_integer_index("depletion_id");
+  else
+    depletion_id = mesh->add_elem_integer("depletion_id");
```

Comment:
```suggestion
  const auto depletion_id = mesh->add_elem_integer("depletion_id");
```

Add will do nothing if it already exists